src/axom/quest/io/STLWriter.hpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_STLWRITER_HPP_
+#define QUEST_STLWRITER_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core/Macros.hpp"
+#include "axom/mint/mesh/Mesh.hpp"
+
+#include <string>
+
+namespace axom
+{
+namespace quest
+{
+/*!
+ * \class STLWriter
+ *
+ * \brief A simple STL writer for the faces of Mint meshes.
+ *
+ * STL (STereoLithography) is a common file format for triangle meshes.
+ * It encodes a "soup of triangles" by explicitly listing the coordinate
+ * positions of the three vertices of each triangle.
+ *
+ * \note Any faces that contain more than 3 vertices will be treated
+ *       like a triangle fan and will result in multiple triangles.
+ *       No care is taken to prevent triangle overlaps for non-convex faces.
+ */
+class STLWriter
+{
+public:
+  /*!
+   * \brief Constructor.
+   */
+  STLWriter() = default;
+
+  /*!
+   * \brief Constructor.
+   *
+   * \param filename The name of the file to write.
+   * \param binary Whether or not to write a binary STL file.
+   */
+  STLWriter(const std::string& filename, bool binary = false);
+
+  /*!
+   * \brief Destructor.
+   */
+  ~STLWriter() = default;
+
+  /*!
+   * \brief Sets the name of the file to write.
+   * \param [in] fileName the name of the file to write.
+   */
+  void setFileName(const std::string& fileName) { m_fileName = fileName; }
+
+  /*!
+   * \brief Sets whether to use binary output.
+   * \param [in] binary True to write binary output; false to write ASCII.
+   */
+  void setBinary(bool binary) { m_binary = binary; }
+
+  /*!
+   * \brief Writes the mesh into an STL file.
+   * \param [in] mesh pointer to the mesh to write.
+   * \pre path to input file has been set by calling `setFileName()`
+   * \return status set to zero on success; set to a non-zero value otherwise.
+   */
+  int write(const mint::Mesh* mesh);
+
+// The following members are protected (unless using CUDA)
+#if !defined(__CUDACC__)
+protected:
+#endif
+
+  /*!
+   * \brief Compute the number of triangles produced for the input mesh.
+   *
+   * \param mesh The input mesh.
+   *
+   * \return The number of triangles.
+   */
+  IndexType getNumberOfTriangles(const mint::Mesh* mesh) const;
+
+  /*!
+   * \brief Determines whether the input mesh looks topologically 2D since
+   *        the mesh dimension seems to reflect the number of coordinate components.
```

Comment:
Minor: I'm confused by this comment. Could you please update it?

Specifically, the word "looks" and the reference to "coordinate components" made me think we were going to check if the mesh was (geometrically) planar.

The check in the implementation is about topological cell types (triangle and quad), not about geometric coordinates, which is in line with the function name.

Reply:
I changed the comments. This functionality was needed when I started using the writer in the mesh_tester. The surface mesh has 3D coordinates. `mint::Mesh::getDimension()` returns 3 (like the coordinates) but there are no faces, just cells, so we have to iterate over the cells rather than using `mint::for_all_faces`.